src/controllers/wallet/wallet.controller.js

```diff
@@ -347,11 +355,55 @@ async function decodeTx(req, res) {
           outputData.decoded = {
             address: output.decodedScript.address.base58,
             timelock: output.decodedScript.timelock,
+            mine: await req.wallet.isAddressMine(output.decodedScript.address.base58)
           };
       }
       data.outputs.push(outputData);
     }
-    res.send({ success: true, tx: data });
+
+    // True if all the inputs are signed, false otherwise
+    data.completeSignatures = data.inputs.length > 0
+      ? data.inputs.every(input => input.signed) // true until find a false statement
+      : false; // empty data.inputs
+
+    // Get balance
+    const balance = {};
+    { // Default HTR balance.
+      const balanceObj = await req.wallet.getBalance(hathorLibConstants.HATHOR_TOKEN_CONFIG.uid);
```

Comment:
This is not the balance of the transaction, it is the balance of the entire wallet.
I think you should use [getTxBalance](https://github.com/HathorNetwork/hathor-wallet-lib/blob/master/src/utils/transaction.ts#L209) instead

Reply:
Maybe I have misunderstood this. I thought I would return wallet metadata as well, not only transaction metadata. That's why I return the wallet balance.